core/iwasm/interpreter/wasm_loader.c

```diff
@@ -827,20 +828,6 @@ load_init_expr(WASMModule *module, const uint8 **p_buf, const uint8 *buf_end,
                                     "constant expression required");
                     goto fail;
                 }
-#else
-                if (global_idx
-                    >= module->import_global_count + module->global_count) {
```

Comment:
need a confirmation. `module->import_global_count + module->global_count` seems doesn't match spec requirement unless there is a different one in GC spec.